src/sdam/server.ts

```diff
@@ -68,52 +80,64 @@ const stateTransition = makeStateMachine({
 
 const kMonitor = Symbol('monitor');
 
+export interface ServerOptions extends ConnectionPoolOptions, ClientMetadataOptions {
+  credentials?: MongoCredentials;
+}
+
+interface ServerPrivate {
+  /** The server description for this server */
+  description: ServerDescription;
+  /** A copy of the options used to construct this instance */
+  options?: ServerOptions;
+  /** A logger instance */
+  logger: Logger;
+  /** The current state of the Server */
+  state: string;
+  /** The topology this server is a part of */
+  topology: Topology;
+  /** A connection pool for this server */
+  pool: ConnectionPool;
+}
+
+export interface DestroyOptions {
+  force?: boolean;
+}
+
 /**
  * @fires Server#serverHeartbeatStarted
  * @fires Server#serverHeartbeatSucceeded
  * @fires Server#serverHeartbeatFailed
  */
 class Server extends EventEmitter {
-  s: any;
-  clusterTime: any;
+  s: ServerPrivate;
```

Comment:
Ah yea I see that its accessed in cmap/wire_protocol, there's no `internal` like swift that'd be nice tho, oh well 
Probably whatever properties we use off `s` elsewhere should be made into getters but we can leave that for follow on work.

Reply:
I see that we have InternalCursorState, maybe we should go with InternalServer / ServerInternal types to imply that same access level, also I see MonitorPrivate that could get similar treatment.
Just bikeshedding, as is is fine.

Reply:
I think ultimately we want to move away from the `XPrivate` types, and move the values directly onto the parent class, so this might not be a problem much sooner.